cypress/support/pageObjects/performancePageLocators.js

```diff
@@ -0,0 +1,335 @@
+            cy.get(this._visitSiteButton)
+                .invoke('removeAttr', 'target') // Prevent opening in a new tab
+                .click();
+    
+            // Wait for the sample page link to be available
+            cy.get('.wp-block-pages-list__item__link.wp-block-navigation-item__content', { timeout: 6000 })
+                .should('be.visible')
+                .invoke('attr', 'href')
+                .then((url) => {
+                    // Intercept the API request with the extracted URL
+                    cy.intercept('GET', url).as('apiRequest');
+    
+                    // Reload the page if necessary
+                    cy.reload(forceReload);
+    
+                    // Navigate to the sample page and wait for the API request to complete
+                    cy.get(this._samplePageButton).click();
+                    cy.wait('@apiRequest');
+    
+                    // Assert the API response status code
+                    cy.get('@apiRequest')
+                        .its('response.statusCode')
+                        .should('eq', statusCode);
+    
+                    // Go back twice
+                    cy.go('back');
+                    cy.go('back');
+                });
+        });
+    }
+
+    interceptCallForMouseDownWithExcludeRunTimeURL(selectedDropDown, requestCount) {
+        const forceReload = true;
+        Cypress.config('defaultCommandTimeout', 4000);
+    
+        // Function to extract page name from URL
+        const extractPageName = (url) => {
+            const pageName = url.split('/').filter(Boolean).pop();
+            cy.log('Extracted page name:', pageName);
+            expect(pageName).to.not.be.empty;
+            return pageName;
+        };
+    
+        // Function to visit the site and check request count
+        const visitSiteAndCheckRequestCount = (url, pageName) => {
+            cy.get(this._excludeKeywordInputField)
+                .clear()
+                .type(pageName);
+    
+            cy.intercept('GET', url).as('apiRequest');
+            cy.get(this._visitSiteButton)
+                .invoke('removeAttr', 'target')
+                .click({ force: true });
+    
+            cy.reload(forceReload);
+            cy.get(this._samplePageButton).click();
+            cy.wrap(requestCount).should('equal', 0);
+    
+            cy.go('back').then(() => {
+                cy.go('back');
+            });
+        };
+    
+        // Function for dropdown interaction logic
+        const handleDropdownSelection = () => {
+            cy.get(this._dropDownForLinkPrefetch).then(($buttonLabel) => {
+                const selectedText = $buttonLabel.text().trim();
+    
+                if (selectedText === selectedDropDown) {
+                    cy.log('Second option is already selected. Proceeding with the test...');
+                    cy.get(this._dropDownForLinkPrefetch).should('have.text', selectedDropDown);
```

Comment:
```suggestion
this._dropDownForLinkPrefetch.invoke('text').should('eq', selectedDropDown);
// else comes here
```
Would it be better to use .should() for assertions and .invoke('text') for checking values instead? This will help us reduce nesting. Similarly look for other cases like these as well.